scikeras/wrappers.py

```diff
@@ -467,35 +463,31 @@ def _check_array_dtype(arr):
             )
         X = check_array(X, allow_nd=True, dtype=_check_array_dtype(X))
 
-        n_features = X.shape[1]
-        target_type = None if y is None else type_of_target(y)
-
-        if reset:
-            self.n_features_in_ = n_features
-            self.target_type_ = target_type
-        else:
-            if n_features != self.n_features_in_:
-                raise ValueError(
-                    f"`X` has {n_features} features, but this {self.__name__}"
-                    f" is expecting {self.n_features_in_} features as input."
-                )
         if y is None:
             return X
         return X, y
 
-    def preprocess_y(
-        self, y: np.ndarray, reset: bool = True
-    ) -> Tuple[
-        Union[np.ndarray, List[np.ndarray], Dict[str, np.ndarray]], Dict[str, Any]
-    ]:
+    def _get_meta(self, X=None, y=None):
+        meta = {}
+        if X is not None:
+            n_features = X.shape[1]
+            meta.update(
+                {"X_dtype_": X.dtype, "X_shape_": X.shape, "n_features_in_": n_features}
+            )
+        if y is not None:
+            target_type = None if y is None else type_of_target(y)
+            meta.update(
+                {"y_dtype_": y.dtype, "y_ndim_": y.ndim, "target_type_": target_type}
```

Comment:
For `X` your are returning `"X_shape_"` but for `y` you are returning `"y_ndim_"`. Is this intentional? It makes sense to me, but I just want to double check.

Reply:
Yes. I checked; only `len(self.y_shape_)` is used. I debated doing the same to `X_shape_`, but feature shape is a lot more important than target shape.

Reply:
I agree. I think `y.ndim` is always going to be 1 or 2 as well. Thanks for clarifying.